frontend/admin/src/js/components/dashboard/Dashboard.tsx

```diff
@@ -151,15 +33,57 @@ const AdminNotFound: React.FC = () => {
   );
 };
 
-export const Dashboard: React.FC<{
-  menuItems: ReactElement[];
+interface OpenMenuButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
+  show: boolean;
+}
+
+const OpenMenuButton: React.FC<OpenMenuButtonProps> = ({
+  show,
+  onClick,
+  children,
```

Comment:
Why use children, if this a component library button? Feels cleaner and more legible at the high level when buttons that don't have another use just get this hidden inside. Thoughts?

I'm not sure there's value in using children so readily? Again, with abstraction, I would pull children out only if we ever come back to re-used this exact button, and not before.

```jsx
      <OpenMenuButton onClick={handleMenuToggle} show={!isMenuOpen}>
        {intl.formatMessage({
          defaultMessage: "Open Menu",
          description: "Text label for header button that opens side menu.",
        })}
      </OpenMenuButton>
```

vs 

```jsx
      <OpenMenuButton onClick={handleMenuToggle} show={!isMenuOpen} />
```

Reply:
i welcome push-back on this btw :)

Reply:
That would require adding `const intl = useIntl();` to the component introducing more complexity when it is just easier using children when `intl` is available in the parent scope. Only reason I chose that direction.